### PR TITLE
RELATED: RAIL-2079 add withMeasures method to IElementsQuery

### PR DESCRIPTION
**Pull request by no23reason**
Add a way to limit the elements by a set of measures.
This limits the elements to only ones for which the set of measures has data.

Also mark restrictiveDefinition and restrictiveDefinitionContent
as deprecated - it is hard to set them correctly
and the withAttributeFilters and withMeasures provide a better alternative.

JIRA: RAIL-2079

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2480/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4093/ : SUCCESS in 15m 42s

File: libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts [hunk 1]
```diff
@@ -41,19 +41,14 @@ export interface IElementsQueryOptions {
     includeTotalCountWithoutFilters?: boolean;
 
     /**
-     * TODO what is this doing?
+     * @deprecated use withAttributeFilters or withMeasures
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts:44]: i say let's just delete these.. i mean.. they did not work anyway so them being here is a bug :)

Comment (no23reason) [reply libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts:44]: I agree, I was just scared 😅

Comment (no23reason) [reply libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts]: Removed in an extra commit

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2481/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4094/ : SUCCESS in 15m 36s

**Review by lupko — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1035

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/739/ : FAILURE in 1m 39s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1002/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4096/ : SUCCESS in 15m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/101/ : SUCCESS in 10m 39s (non-voting)

**Pull request merged**